src/pages/SearchPage/SearchPage.js

```diff
@@ -17,25 +17,53 @@ async function handleSearch(subreddit) {
 
 function SearchPage() {
   const [isLoading, setIsLoading] = useState(true);
+  const [dataLoaded, setDataLoaded] = useState(false);
   const [searchResults, setSearchResults] = useState([]);
+  const [heatmapInfo, setHeatMapInfo] = useState([]);
   const { subreddit } = useParams();
 
+  function parseRedditData() {
+    const arrInfo = Array(7).fill(0).map(() => Array(24).fill(0));
+    const results = searchResults.data;
+    for (let i = 0; i < results.length; i += 1) {
+      // Add utc seconds from API to base epoch time
+      const date = new Date(0);
+      date.setSeconds(results[i].created_utc);
+      const day = date.getDay();
+      const hour = date.getHours();
+      arrInfo[day][hour] += 1;
+    }
+    setHeatMapInfo(arrInfo);
+  }
+
   // Search for new subreddit when parameters have changed
   useEffect(() => {
+    setDataLoaded(false);
```

Comment:
This flag is used to distinguish between "everything is ok" and "an error occurred", right? I'd rather use a `setError` state here, since you need to show the error anyways. You could then simply save a string to the error state and render that instead of the spinner and heatmap

Reply:
I want to implement this but I'm not sure exactly what you mean. If it's not too much trouble, would you be able to expand on what you mean by saying to `save a string to the error state and render that instead of the spinner and heatmap`?

Thank you

Reply:
Sure. I mean instead of

```
const [dataLoaded, setDataLoaded] = useState(false);
setDataLoaded(false);
```

You could

```
const [error, setError] = useState(null);
setError(error.message);
// or simply
setError('An error occurred. Please try again')
```

Then you can also display the error instead of the heatmap like this or so

```
error && <div>{error}</div>
```